libtorch/release.nix

```diff
@@ -3,93 +3,55 @@
 with pkgs;
 
 let
-  callCpu = opts: callPackage ./generic.nix ({ mklSupport = true; } // opts);
+  libmklml = opts: callPackage ./mklml.nix ({
+  } // opts);
+  callCpu = opts: callPackage ./generic.nix ({
+    mklml = libmklml {};
+    libcxx = libcxx;
+  } // opts);
   callGpu = opts: callPackage ./generic.nix ({
-    mklSupport = true;
-    cudaSupport = true;
+    mklml = libmklml {};
+    libcxx = libcxx;
   } // opts);
 in
-
 {
+  libmklml = libmklml {};
   libtorch_cpu = callCpu {
-    version = "1.1";
+    version = "1.2";
+    buildtype = "cpu";
     mkSrc = buildtype:
       if stdenv.hostPlatform.system == "x86_64-linux" then
         fetchzip {
-          url = "https://download.pytorch.org/libtorch/${buildtype}/libtorch-shared-with-deps-latest.zip";
-          sha256 = "09iwdy31zg9dzkrjx8mwpds9mxrv775msn01v1njkpjymvi7llz6";
+          url = "https://download.pytorch.org/libtorch/cpu/libtorch-cxx11-abi-shared-with-deps-1.2.0.zip";
+          sha256 = "0vh5fw9h1rydp2bbrlkq54z29p1v0lpfmllddk82sgz7sr3jld66";
         }
       else if stdenv.hostPlatform.system == "x86_64-darwin" then
         fetchzip {
-          url = "https://download.pytorch.org/libtorch/${buildtype}/libtorch-macos-1.1.0.zip";
-          sha256 = "03wqgvmyz2dv5iin27rnhxy6blk7gf0h49vgwmnab9c5j43y2y3d";
+          url = "https://download.pytorch.org/libtorch/cpu/libtorch-macos-1.2.0.zip";
+          sha256 = "0qglhy7dpjxcn24q41wp0n8dflypbmfk6mqzafavi2jfhl33wac2";
         }
       else throw "missing url for platform ${stdenv.hostPlatform.system}";
   };
-  libtorch_cudatoolkit_10_0 = callGpu {
-    version = "1.1";
+  ${if stdenv.hostPlatform.system == "x86_64-darwin" then null else "libtorch_cudatoolkit_10_0"} = callGpu {
```

Comment:
that's some nix fu right there. does `null` on the left-hand side mean that the right-hand side is assigned to nothing and hence cannot be referenced?

Reply:
Yes. This is to test this file by using "nix-build -E '((import <nixpkgs>) {}).callPackages ./release.nix {}' ''.
Should I use just "libtorch_cudatoolkit_10_0" not using null?

Reply:
I like it, I didn’t even know this was possible. Well done!